src/sentry/api/endpoints/auth_register.py

```diff
@@ -0,0 +1,140 @@
+import logging
+
+from django.conf import settings
+from django.core.exceptions import ValidationError as DjangoValidationError
+from django.db import transaction
+from rest_framework import serializers
+
+from sentry import newsletter, options
+from sentry.api.base import Endpoint
+from sentry.api.serializers.base import serialize
+from sentry.api.serializers.rest_framework.base import CamelSnakeSerializer
+from sentry.app import ratelimiter
+from sentry.auth import password_validation
+from sentry.models import User
+from sentry.signals import user_signup
+from sentry.utils import auth, metrics
+
+logger = logging.getLogger("sentry.auth")
+audit_logger = logging.getLogger("sentry.audit.ui")
+
+ERR_UNKNOWN = "An unknown error occurred while creating the account."
+
+
+def send_confirm_emails(user):
+    for email in user.emails.filter(is_verified=False):
+        user.send_confirm_email_singular(email=email, is_new_user=True)
+
+
+class AuthRegisterSerializer(CamelSnakeSerializer):
```

Comment:
this does not implement the org creation logic, as its fairly complex, and its simpler for us to do that in our override in getsentry given we want to use a custom code path (provision)